pipeline/api/cloud.py

```diff
@@ -287,17 +307,35 @@ def upload_pipeline(
 
         new_variables: List[PipelineVariableGet] = []
         print("Uploading variables")
+
+        from pipeline.objects import PipelineFile
+
         for _var in new_pipeline_graph.variables:
             _var_type_file = self.upload_file(
                 io.BytesIO(python_object_to_hex(_var.type_class).encode()), "/"
             )
+
+            pipeline_file_schema: PipelineFileVariableGet = None
+
+            if isinstance(_var, PipelineFile):
+
+                _var_file_hash = self._hash_file(_var.path)
+
+                _var_file = self.upload_file(_var.path, "/")
```

Comment:
not sure how large these files are but if they could be very large then perhaps this could be made more efficient since it seems like we're reading the contents of the file twice (once to compute the hash and once in `upload_file`)?

Reply:
Hashing a file was fast on my computer (hundreds of ms for 500MB) and the idea is to check the hash with the API in the future to check if it has been uploaded previously. Unfortunately it will mean having to open it twice.

Reply:
Maybe I didn't explain well but I just meant that the `upload_file()` and `_hash_file()` are both reading in the contents of the file. So one optimisation could be to do something like (in pseudo-code):
```python
file_contents = read_file(file_path)
hash_file(file_contents)
upload_file(file_contents)
```

It might be that reading in the contents of the file is pretty fast though so maybe not worth worrying about

Reply:
Okay gotcha - yeah I think it's best to leave this for the moment actually. The issue is the upload method for large files, imagine the file is 10GB like GPT J we don't want to load the whole thing into memory and then send. We want to send chunks of it, and the hash function doesn't load the whole thing thankfully.

Reply:
Cool, yeah we might want to update the `upload_file` method to support uploading in chunks rather than the whole file at some point too